Troubleshoot/TroubleshootError.ps1

```diff
@@ -231,7 +262,104 @@ else {
+                Write-Host("");
+            }
+            else {
+                $TryToOnboardToCustomMetrics = "false";
+                if ($MonitoringMetricsPublisherCandidates) {
+                    Write-Host($MonitoringMetricsPublisherCandidates | Format-List | Out-String);
+
+                    $totalCandidates = $MonitoringMetricsPublisherCandidates.ObjectId.Length;
+                    $metricsPublisherRoleAlreadyExists = "false";
+
+                    for ($index = 0; $index -lt $totalCandidates; $index++) {
+                        if ($MonitoringMetricsPublisherCandidates.ObjectId[$index] -eq $clusterSPNObjectID) {
+                            $metricsPublisherRoleAlreadyExists = "true";
+                        }
+                    }
+
+                    if ($metricsPublisherRoleAlreadyExists -eq "true") {
+                        Write-Host("Cluster SPN has the Monitoring Metrics Publisher Role assigned already") -ForegroundColor Green;
+                    }
+                    else {
+                        $TryToOnboardToCustomMetrics = "true";
+                    }
+                }
+                else {
+                    Write-Host("No monitoring metrics publisher candidates present, We need to onboard the cluster service prinicipal to the Monitoring Metrics Publisher role");
+                    $TryToOnboardToCustomMetrics = "true";
+                }
+                if ($TryToOnboardToCustomMetrics -eq "true") {
+                    $message = "Detected that custom metrics is not enabled for this cluster. You need to be an owner on the cluster resource to do the following operation operation.";
+                    Write-Host($message);
+                    $question = " Do you want this script to enable it by adding the role 'Monitoring Metrics Publisher' to your clusters SPN?"
+
+                    $choices = New-Object Collections.ObjectModel.Collection[Management.Automation.Host.ChoiceDescription]
+                    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Yes'))
+                    $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&No'))
+
+                    $decision = $Host.UI.PromptForChoice($message, $question, $choices, 0);
+
+                    if ($decision -eq 0) {
+                        $AssignRoleAssignment = New-AzureRmRoleAssignment -ObjectId $clusterSPNObjectID -Scope $AKSClusterResourceId -RoleDefinitionName $MonitoringMetricsRoleDefinitionName -ErrorAction SilentlyContinue -ErrorVariable assignmentFailed;
+                        if ($assignmentFailed) {
+                            Write-Host("Couldn't assign the new role. You need the cluster owner role to do this action. Please contact your cluster administrator to onboard.") -ForegroundColor Red;
+                            Write-Host("Please contact us by emailing askcoin@microsoft.com for help") -ForegroundColor Red;
```

Comment:
I think, we just point to the doc how to resolve the permission issue.

Reply:
which doc are you referring to?